docs/docs/reconciliation.md

```diff
@@ -73,61 +67,56 @@ renderB: <div style={{'{{'}}fontWeight: 'bold'}} />
 
 After the attributes have been updated, we recurse on all the children.
 
-
 ### Custom Components
 
-We decided that the two custom components are the same. Since components are stateful, we cannot just use the new component and call it a day. React takes all the attributes from the new component and calls `componentWillReceiveProps()` and `componentWillUpdate()` on the previous one.
+The last case is comparing two custom components of the same type. Since components are stateful, we must keep the old instance around. React takes all the attributes from the new component and calls `componentWillReceiveProps()` and `componentWillUpdate()` on the previous one.
```

Comment:
Hmmmm, what about the case where one is a DOM element and one is a component element? Like if you have a component ShimButton which always just renders to a <button/>, and then if you have a second component SpitefulButton which first renders to <ShimButton/>, and then at some time later rerenders directly to <button/>. React will just throw away the first tree and rerender the <button/> element... right?